base/services/w32time/w32time.c

```diff
@@ -0,0 +1,418 @@
+                         &hToken))
+    {
+        priv.PrivilegeCount = 1;
+        priv.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
+
+        if (LookupPrivilegeValueW(NULL,
+                                  SE_SYSTEMTIME_NAME,
+                                  &priv.Privileges[0].Luid))
+        {
+            if (AdjustTokenPrivileges(hToken,
+                                      FALSE,
+                                      &priv,
+                                      sizeof(previouspriv),
+                                      &previouspriv,
+                                      &PrevSize) &&
+                GetLastError() == ERROR_SUCCESS)
+            {
+                /*
+                 * We successfully enabled it, we're permitted to change the time.
+                 */
+                Ret = SetSystemTime(lpSystemTime);
+
+                /*
+                 * For the sake of security, restore the previous status again
+                 */
+                if (previouspriv.PrivilegeCount > 0)
+                {
+                    AdjustTokenPrivileges(hToken,
+                                          FALSE,
+                                          &previouspriv,
+                                          0,
+                                          NULL,
+                                          0);
+                }
+            }
+        }
+        CloseHandle(hToken);
+    }
+
+    return Ret;
+}
+
+
+/*
+ * NTP servers state the number of seconds passed since
+ * 1st Jan, 1900. The time returned from the server
+ * needs adding to that date to get the current Gregorian time
+ */
+static DWORD
+UpdateSystemTime(ULONG ulTime)
+{
+    FILETIME ftNew;
+    LARGE_INTEGER li;
+    SYSTEMTIME stNew;
+
+    /* Time at 1st Jan 1900 */
+    stNew.wYear = 1900;
+    stNew.wMonth = 1;
+    stNew.wDay = 1;
+    stNew.wHour = 0;
+    stNew.wMinute = 0;
+    stNew.wSecond = 0;
+    stNew.wMilliseconds = 0;
+
+    /* Convert to a file time */
+    if (!SystemTimeToFileTime(&stNew, &ftNew))
+    {
+        return GetLastError();
+    }
+
+    /* Add on the time passed since 1st Jan 1900 */
+    li = *(LARGE_INTEGER *)&ftNew;
+    li.QuadPart += (LONGLONG)10000000 * ulTime;
+    ftNew = * (FILETIME *)&li;
+
+    /* Convert back to a system time */
+    if (!FileTimeToSystemTime(&ftNew, &stNew))
+    {
+        return GetLastError();
+    }
+
+    if (!SystemSetTime(&stNew))
+    {
+        return GetLastError();
+    }
+
+    return ERROR_SUCCESS;
+}
+
+
+static DWORD
+GetIntervalSetting(VOID)
+{
+    HKEY hKey;
+    DWORD dwData;
+    DWORD dwSize = sizeof(dwData);
+    LONG lRet;
+
+    dwData = 0;
+    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                      L"SYSTEM\\CurrentControlSet\\Services\\W32Time\\TimeProviders\\NtpClient",
+                      0,
+                      KEY_QUERY_VALUE,
+                      &hKey) == ERROR_SUCCESS)
+    {
+        /* This key holds the update interval in milliseconds
```

Comment:
This should say "seconds" instead of "milliseconds" and then "10" instead of "10000" on the next statement.